lib/reline/line_editor.rb

```diff
@@ -112,7 +112,11 @@ def set_pasting_state(in_pasting)
     else
       prompt = @prompt
     end
-    if @prompt_proc
+    if !@is_multiline
+      mode_string = check_mode_string
+      prompt = mode_string + prompt if mode_string
+      [prompt] + [''] * (buffer.size - 1)
```

Comment:
In Readline, prompt for multiline input looks like this
```
irb(main):001> if 1
2
3
4
end
```

This change will fix IRB's show_doc interactive search prompt bug
```
irb(main):001> show_doc

Enter the method name you want to look up.
You can use tab to autocomplete.
Enter a blank line to exit.

irb(main):001>  # this line should be ">> "
```

Reply:
This will be amazing 🎉 
Is it possible to cover this fix with a test?

Reply:
Added 👍